internal/openshiftcontroller/openshiftcontroller.go

```diff
@@ -25,49 +25,27 @@ type OpenShiftController struct {
 	Conditions       Conditions
 	Users            map[string]*User
 	lock             *sync.Mutex
-	groupLock        *sync.Mutex
-	Groups           []*[]string
-	groupSleep       time.Duration
+	checkIdleSleep   time.Duration
```

Comment:
👍, next step is to change this a bit more and get away from the loop iterating over all users and then going to sleep. I want to end up in a model where we have one go routine per user which listens on a timer channel as well as on a channel created by the watch routine.